crates/hc_sandbox/src/cmds.rs

```diff
@@ -93,80 +93,100 @@ pub enum NetworkType {
 
 #[derive(Debug, Parser, Clone)]
 pub struct Existing {
-    /// Paths to existing sandbox directories.
-    /// For example `hc sandbox run -e=/tmp/kAOXQlilEtJKlTM_W403b,/tmp/kddsajkaasiIII_sJ`.
-    #[arg(short, long, value_delimiter = ',')]
-    pub existing_paths: Vec<PathBuf>,
-
     /// Run all the existing conductor sandboxes specified in `$(pwd)/.hc`.
-    #[arg(short, long, conflicts_with_all = &["last", "indices"])]
+    #[arg(short, long, conflicts_with_all = &["indices"])]
     pub all: bool,
 
-    /// Run the last created conductor sandbox --
-    /// that is, the last line in `$(pwd)/.hc`.
-    #[arg(short, long, conflicts_with_all = &["all", "indices"])]
-    pub last: bool,
-
     /// Run a selection of existing conductor sandboxes
     /// from those specified in `$(pwd)/.hc`.
     /// Existing sandboxes and their indices are visible via `hc list`.
     /// Use the zero-based index to choose which sandboxes to use.
     /// For example `hc sandbox run 1 3 5` or `hc sandbox run 1`
-    #[arg(conflicts_with_all = &["all", "last"])]
+    #[arg(conflicts_with_all = &["all"])]
     pub indices: Vec<usize>,
 }
 
 impl Existing {
-    pub fn load(mut self) -> anyhow::Result<Vec<PathBuf>> {
-        let sandboxes = crate::save::load(std::env::current_dir()?)?;
-        if self.all {
-            // Get all the sandboxes
-            self.existing_paths.extend(sandboxes)
-        } else if self.last && sandboxes.last().is_some() {
-            // Get just the last sandbox
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
+    pub fn load(self, hc_dir: PathBuf) -> std::io::Result<Vec<PathBuf>> {
+        let sandboxes = crate::save::load(hc_dir)?;
+        if sandboxes.is_empty() {
```

Comment:
Check for empty list first now. Otherwise loading an invalid dir with Existing::all would not prompt an error and message to the user.